docs/book/src/cronjob-tutorial/testdata/project/controllers/suite_test.go

```diff
@@ -0,0 +1,154 @@
+/*
+When you created the CronJob API with `kubebuilder create api`, Kubebuilder already did some test work for you.
+Kubebuilder generates a controllers/suite_test.go file that does the bare bones of setting up a test environment.
+Your automatically generated file will contain multiple helpful components.
+
+First, it will contain necessary imports.
+These primarily pull in the Gingko and Gomega testing frameworks and the envtest k8s testing package.
+These imports also make available the k8s client you’ll use in your testing, and import the API you build to expose your CronJob Kind.
+*/
+
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// +kubebuilder:docs-gen:collapse=Apache License
+
+package controllers
+
+import (
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	batchv1 "tutorial.kubebuilder.io/project/api/v1"
+
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	// +kubebuilder:scaffold:imports
+)
+
```

Comment:
The cron-job example is outdated. 
However, the suit_test.go is gen automatically. IMO should use sem file that is generated and just add the explanations. See: https://github.com/kubernetes-sigs/kubebuilder/blob/d6e2b995831ce26228f2303e9d1b3a55acd3d225/testdata/project-v3/controllers/suite_test.go

Reply:
Gotcha. Does this mean I would also have to re-write the example unit test to now support this new example instead of the cronjob one? Are there plans to roll over to using this new example in the docs any time soon?

Reply:
Hi @gabbifish I created the PR: https://github.com/kubernetes-sigs/kubebuilder/pull/1492 to try updated the project with the latest versions. Also, we have an issue to follow up all docs and try to update some outdated point found. See; https://github.com/kubernetes-sigs/kubebuilder/issues/1251

However, IMO it is fine to move forward here, we just need to ensure that this file is as generated by the master branch changes and with your changes on top. Is it make sense? Could I explain properly my suggestion?

Reply:
^ This makes sense! The `suite_test.go` code in this example is consistent with the autogenerated code provided by `kubebuilder create api`. I did move 
```go
func TestAPIs(t *testing.T) {
	RegisterFailHandler(Fail)

	RunSpecsWithDefaultAndCustomReporters(t,
		"Controller Suite",
		[]Reporter{envtest.NewlineReporter{}})
}
``` 
underneath the `Before` and `After` function calls because I thought this ordering might be easier for new users to read through, but am happy to put this function in its original position if you think that would be better!

Reply:
Fell free to close this one .. all solved